lib/symboldatabase.cpp

```diff
@@ -4753,9 +4753,9 @@ const Scope *Scope::findRecordInBase(const std::string & name) const
                     return base->classScope;
                 }
 
-                const ::Type * type = base->classScope->findType(name);
-                if (type)
-                    return type->classScope;
+                const ::Type * t = base->classScope->findType(name);
```

Comment:
hmm. I have the feeling that it would be good if Cppcheck also warned about these.

Reply:
I will add a ticket for this soon.